### PR TITLE
Deprecate Empty Templates

### PR DESCRIPTION
**Pull request by jdconrad**
Deprecates empty templates.  This will be followed up with another PR for just 7.0 that will remove the ability to add empty templates, and will drop empty templates during upgrades with a warning.

Also removes a backwards incompatible check for empty code strings being loaded from stored scripts. Search templates did not have an appropriate check for not allowing stored scripts to have an empty code string, so when an empty search template is added, future restarts of the node will fail.

### CODE REVIEW

**Comment by jdconrad** [comment]
@martijnvg Hey Martijn, if you're back from holiday, would appreciate you having a look at this when you have a spare moment.  Thanks!

**Comment by nik9000** [comment]
5.6.10 seems like a long way to backport this. If you plan from removing from master I think it is fine to release this with 6.4.

File: server/src/main/java/org/elasticsearch/script/StoredScriptSource.java [hunk 1]
```diff
@@ -131,18 +138,37 @@ private void setOptions(Map<String, String> options) {
 
         /**
          * Validates the parameters and creates an {@link StoredScriptSource}.
+         *
+         * @param ignore Specify as {@code true} to ignore the empty source check.  This allows
+         *         empty templates to be loaded for backwards compatibility.
          */
-        private StoredScriptSource build() {
+        private StoredScriptSource build(boolean ignore) {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/script/StoredScriptSource.java]: `s/ignore/ignoreEmpty/` maybe?

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by jdconrad** [comment]
@martijnvg @nik9000 Thanks for the reviews!  Will push as soon as CI is completed.

**Pull request merged**